### ISSUE TITLE
Loading Incompatible Save Ruins New Game

### ISSUE THREAD

**Issue by benskywalker** [C7-Game/Prototype#434]
Data loaded from an incompatible save persists until the program is restarted. After the error message, this data will still be in the game if you start a new game.

![image](https://github.com/C7-Game/Prototype/assets/12992619/b933c6af-a6cf-435e-b998-3c2010c627db)
Here I have altered terrain from the scenario I loaded. However, the units can not be moved due to incompatibilities

**Comment by pcen** [comment]
What data is persisting? I'm working on revamping the way saves are handled in general, which should address a lot of save-related issues in general. Currently saves are quite brittle. There's some context in [this discussion](https://github.com/C7-Game/Prototype/discussions/377)

**Comment by benskywalker** [comment]
> What data is persisting? I'm working on revamping the way saves are handled in general, which should address a lot of save-related issues in general. Currently saves are quite brittle. There's some context in [this discussion](https://github.com/C7-Game/Prototype/discussions/377)

I can do more testing, but the only thing I found so far is the terrain images. The image above shows the altered terrain used in the scenario I _tried_ to load, which failed. When I then hit "Quick Start" without closing the game, I had the altered terrain from the scenario but I could not move my units

**Comment by pcen** [comment]
Which branch does this happen on?

**Comment by benskywalker** [comment]
Good question, sorry for being vague. I didn't remember so I just tested it again. I'm on use-tilemaps now. The game actually runs fine but has altered terrain graphics! Closing the issue but it's worth noting that some data is maintained from an aborted load of a scenario (only terrain graphics as far as I can tell).
![image](https://github.com/C7-Game/Prototype/assets/12992619/2843db0c-dd78-4db3-9807-675e114ca5c8)
